tools/state-viewer/src/replay_headers.rs

```diff
@@ -67,23 +73,143 @@ pub(crate) fn replay_headers(
+            println!(
+                "{}: Stake [diff={}] {} --> {}",
+                account_id.as_str(),
+                right_summary.stake - left_summary.stake,
+                left_summary.stake,
+                right_summary.stake
+            );
+        }
+        if left_summary.block_production_rate != right_summary.block_production_rate {
+            println!(
+                "{}: Blocks [diff=%{}] %{} --> %{}",
+                account_id.as_str(),
+                right_summary.block_production_rate - left_summary.block_production_rate,
+                left_summary.block_production_rate,
+                right_summary.block_production_rate
+            );
+        }
+        if left_summary.chunk_production_rate != right_summary.chunk_production_rate {
+            println!(
+                "{}: Chunks [diff=%{}] %{} --> %{}",
+                account_id.as_str(),
+                right_summary.chunk_production_rate - left_summary.chunk_production_rate,
+                left_summary.chunk_production_rate,
+                right_summary.chunk_production_rate
+            );
+        }
+        if left_summary.chunk_endorsement_rate != right_summary.chunk_endorsement_rate {
+            println!(
+                "{}: Endorsements [diff=%{}] %{} --> %{}",
+                account_id.as_str(),
+                right_summary.chunk_endorsement_rate - left_summary.chunk_endorsement_rate,
+                left_summary.chunk_endorsement_rate,
+                right_summary.chunk_endorsement_rate
+            );
+        }
+    }
+}
+
+fn compare_validator_kickout_stats(left: &EpochValidatorInfo, right: &EpochValidatorInfo) {
+    let left_kickouts = get_validator_kickouts(left);
+    let right_kickouts = get_validator_kickouts(right);
+    let left_accounts: HashSet<AccountId> = left_kickouts.keys().cloned().collect();
+    let right_accounts: HashSet<AccountId> = right_kickouts.keys().cloned().collect();
+    for account_id in left_accounts.union(&right_accounts) {
+        let left_kickout = left_kickouts.get(account_id);
+        let right_kickout = right_kickouts.get(account_id);
+        if left_kickout.is_some() ^ right_kickout.is_some() {
+            println!("{}: {:?} --> {:?}", account_id.as_str(), left_kickout, right_kickout);
+            continue;
+        }
+        let left_kickout = left_kickout.unwrap();
+        let right_kickout = right_kickout.unwrap();
+        if left_kickout != right_kickout {
+            println!("{}: {:?} --> {:?}", account_id.as_str(), left_kickout, right_kickout);
+        }
+    }
+}
+
+/// Returns a mapping from validator account id to the summary of production statistics.
+fn get_validator_summary(
+    validator_info: &EpochValidatorInfo,
+) -> HashMap<AccountId, ValidatorSummary> {
+    let mut summary = HashMap::new();
+    for validator in validator_info.current_validators.iter() {
+        summary.insert(
+            validator.account_id.clone(),
+            ValidatorSummary {
```

Comment:
nit: Would be more readable in multiple lines

Reply:
done